src/TableOfContents/index.tsx

```diff
@@ -19,6 +19,8 @@ export type TableOfContentsItem = {
   icon?: FAIconProp;
   activeIcon?: FAIconProp;
   iconColor?: string;
+  iconPosition?: 'left' | 'right';
+  textIcon?: string;
```

Comment:
It's starting to feel like our abstraction is broken: there's way too many `icon` props allowing for not too many use-cases.
Not something to block this PR, but next time we touch ToC this will likely need some cleanup.